cpu/esp_common/esp-now/esp_now_netdev.c

```diff
@@ -452,6 +453,24 @@ esp_now_netdev_t *netdev_esp_now_setup(void)
     return dev;
 }
 
+int esp_now_set_channel(uint8_t channel)
+{
+#ifdef ESP_NOW_UNICAST
+    scan_cfg.channel = channel;
+#endif
+#ifndef MODULE_ESP_WIFI
+    wifi_config_ap.ap.channel = channel;
+    esp_err_t result = esp_wifi_set_config(ESP_IF_WIFI_AP, &wifi_config_ap);
+    if (result != ESP_OK) {
+        LOG_TAG_ERROR("esp_now",
+                      "esp_wifi_set_config softap failed with return value %d\n",
+                      result);
+        return result;
+    }
+#endif
+    return ESP_OK;
```

Comment:
Only if `ESP_NOW_UNICAST` is defined or `MODULE_ESP_WIFI` is not defined, otherwise `return -ENOTSUP;`

Reply:
BTW, I already thought about removing the `esp_now` multicast variant because it is not working well since WiFi multcasts are unreliable. The packet loss is about 30 %. The advantage of the multicast variant is that periodic scanning and pairing with other ESP-NOW nodes is not necessary.

Reply:
> Only if `ESP_NOW_UNICAST` is defined or `MODULE_ESP_WIFI` is not defined, otherwise `return -ENOTSUP;`

Ok, my logic behind the function was a little different. I defined the return value from `esp_wifi`'s point of view, which uses the function to inform `esp-now` about the used channel. But from user's point of view who is trying to set the channel, you are right. 

I changed the logic a bit.